.github/guides/STYLE.md

```diff
@@ -399,6 +400,192 @@ turn_on(power_usage = 30) // Fine!
 set_invincible(FALSE) // Fine! Boolean parameters don't always need to be named. In this case, it is obvious what it means.
 ```
 
+## Macros
+
+Macros are, in essence, direct copy and pastes into the code. They are one of the few zero cost abstractions we have in DM, and you will see them often.
+
+This section will assume you understand the following concepts:
+
+### Language - Hygienic
+We say a macro is [**hygienic**](https://en.wikipedia.org/wiki/Hygienic_macro) if, generally, it does not rely on input not given to it directly through the call site, and does not effect the call site outside of it in a way that could not be easily reused somewhere else.
+
+An example of a non-hygienic macro is:
+
+```dm
+#define GET_HEALTH(health_percent) (##health_percent * max_health)
```

Comment:
if you're gonna use the ## pattern it'd be good to explain why, so it doesn't confuse people. Perhaps also link to the byond ref on define?

Reply:
I'll throw it in